#462 & #352 Fixed FPS limiter

Rekperion opened the pull request:
FPS limiter works as intended. Uses VSync when not given an FPS limit as well as fixes several bugs with the FPS limiter that was causing specific input values to make the limit uncapped. FPS limit now also persists into menus once set and is constant for the time the game is running. Unfortunately don't have it so the FPS limit set is saved as I don't know how the prefabs work and I don't want to hold up this issue going through when it can be very beneficial for testers on lower end systems.

Rekperion committed:
#462 Fixed issue where FPS limiter stopped working

The removal of the Serialize stuff fixed the FPS limiter to work again and the FPS counter updates after the pause menu is closed. It doesn't update right away because of how the game pausing actually works but the limiter works perfect now!

Rekperion commented:
First edits seemed to work when testing through unity but after I made a build the issue was still present.

Rekperion commented:
After further investigation it appears this fixes it only when not playing full screen. This means the issue could stem from the full screen settings as well as several other issues could only be unique to the full screen view.

Rekperion committed:
Update FPSLimiter.cs

Rekperion committed:
Update FPSLimiter.cs

Rekperion committed:
Update FPSLimiter.cs

This update to the FPS limiter ensures that the FPS limit can be changed at any time and upon the removal of the limit the FPS of the game will not go above base system refresh rates as what was happening earlier.

Rekperion commented:
After my personal testing it appears that the FPS just doesn't want to update sometimes and other times it will update all the time without fail. From my testing if the fps cap worked once it will continue to work no matter the change but other times it doesn't update. Still need to investigate where the issue lies but the FPS limiter itself at this point works like a beauty and can be easily modified in the future if needed for new features or changes.

Rekperion committed:
Update FPSLimiter.cs

Rekperion commented:
Much more personal testing and I have found that another system is taking priority over the FPS limiter in the builds as everything works wonderfully in a unity testing environment where as with a build the fps limit doesn't update even if the resolution is changed. Odds are whatever is in control of the VSync is causing this issue.

Rekperion committed:
Removed hard coded VSync

Removal of the hard code in the start function allows for the FPS limiter to be used again.

Rekperion commented:
The issue was the FengManager having the hard code to always go off the system specs. I have it so the FPS Limiter will default to the system specs if the user does not otherwise have an FPS limit set

Rekperion commented:
Going through and making tweaks based off tester feedback now. Should be fully done soon.

Rekperion committed:
Update FPSLimiter.cs

Will no longer take negative values that resulted in an uncapped fps.

Rekperion commented:
SOON TO BE DONE!

Rekperion commented:
My bad on the bad branch name. I'll have it more streamlined for all work moving foreword.

Rekperion committed:
Update FPSLimiter.cs

Rekperion committed:
Added some more comments

Lithrun commented:
Hm, not too sure why the CI keeps failing here. Will try to merge development into this branch, and then see if things still go wrong

Lithrun commented:
Ah wait, nvm, it's because it's a fork

Lithrun commented on Assets/Scripts/FengGameManagerMKII.cs:
```diff
@@ -2741,9 +2741,6 @@ public void spawnPlayerAtRPC(float posX, float posY, float posZ, PhotonMessageIn
         
         private void Start()
         {
-            QualitySettings.vSyncCount = 1;
```
Removing these lines of codes causing the game to run without a FPS limit when you are starting the game. This caused #351

Rekperion replied:
should be good to go now!

Rekperion replied:
I only got rid of the "QualitySettings.vSyncCount = 1;" reference as this will permanently lock the fps with VSync, Which resulted in the limiter not being able to be used at all. Setting the fps upon start in the manager is fine on it's own as this does not lock the FPS and block the FPSLimiter from working.

Lithrun reviewed (changes requested):
See the review comment. Other than that, try to reduce the amount of comments you are using on your code. Always use /// summary documentation instead of a comment above a method, and also ask yourself if this comment is really necessary. "SetFPSLimit" would mean that this method sets the Frame Limit, so the comment above it is redundant.

Also avoid using built-in type structs, and instead use their C# keyword (i.e. Instead of Int32, use int. Instead of String, use string. Instead of Boolean, use bool ect.)

Rekperion committed:
Update FPSLimiter.cs

Rekperion committed:
Update FPSLimiter.cs

Rekperion committed:
Update FengGameManagerMKII.cs

This will turn on vsync on the start. The "QualitySettings.vSyncCount = 1;" hard locks the whole game always to be in vsync but with this we can have the menu capped without affecting the FPSLimitter.

Rekperion commented:
I will make sure these changes work on the latest version as soon as I can!

Rekperion committed:
Updated manager changes with new dev updates

Rekperion commented:
Needs to be updated with new changes. Will be done in the next couple of days.

Rekperion commented:
update to PR will be coming thursday/friday

Rekperion committed:
Add files via upload

Rekperion committed:
Add files via upload

Rekperion committed:
Add files via upload
